vms/platformvm/state/state.go

```diff
@@ -674,31 +542,23 @@ func (s *state) GetSubnets() ([]*txs.Tx, error) {
 
 func (s *state) AddSubnet(createSubnetTx *txs.Tx) {
 	s.addedPermissionedSubnets = append(s.addedPermissionedSubnets, createSubnetTx)
+	if s.permissionedSubnetCache != nil {
+		s.permissionedSubnetCache = append(s.permissionedSubnetCache, createSubnetTx)
```

Comment:
`s.permissionedSubnetCache` has to be updated, or else `GetSubnets` may return a stale result after the first call.